### ISSUE TITLE
MailDataRequired requires MailContent?

### ISSUE THREAD

**Issue by 0x80** [sendgrid/sendgrid-nodejs#1057]
### Issue Summary
I'm running into a problem with Typescript because it doesn't let me build an email without a content field. Is the `MailDataRequired` correct?

I never used to send content with my emails. Only substitutions. The content / templates live on your servers.

### Code Snippet
```ts
export type MailDataRequired = MailData & (
    { text: string } | { html: string } | { templateId: string } | { content: MailContent[] & { 0: MailContent } });
```
### Exception/Log
```
# paste exception/log here
```

### Technical details:
* sendgrid-nodejs version: @sendgrid/client@^6.5.3

**Comment by 0x80** [comment]
BTW: 6.5.0 doesn't seem to have this weird `{ 0: MailContent }` requirement, but still content is required.

**Comment by jonbcampos-alto** [comment]
watching this one too. updated my package and now not sure about this

**Comment by childish-sambino** [comment]
Considering this a duplicate of https://github.com/sendgrid/sendgrid-nodejs/issues/1056, unless I'm mistaken.

#1041 introduced in the 6.5.4 release required one of `text`, `html`, or `content`.  It was later pointed out that `templateId` by itself should also be allowed. This was fixed as part #1053 but has yet to be released.

**Comment by rodrigomf24** [comment]
@childish-sambino this is still an issue on `6.5.4` the type is defined like this:

```
type MailDataRequired = MailData & (
  { text: string } | { html: string } | { content: MailContent[] & { 0: MailContent } });
```

That is making `content` a required property, is there a reason for that when `MailData` defines it as a conditional property?  I fixed it by passing:

```
{...mailData, content: undefined}
```

**Comment by childish-sambino** [comment]
> #1041 introduced in the 6.5.4 release required one of `text`, `html`, or `content`.

@rodrigomf24 Does `mailData` not have one of those?

**Comment by rodrigomf24** [comment]
@childish-sambino  no just a `templateId` and `dynamicTemplateData`

**Comment by childish-sambino** [comment]
@rodrigomf24 
>  It was later pointed out that templateId by itself should also be allowed. This was fixed as part #1053 but has yet to be released.

**Comment by saveraa** [comment]
Is this fix released?

**Comment by saveraa** [comment]
I just got version 7.0.0 and I still see this error 

`TSError: ⨯ Unable to compile TypeScript:
src/email-templates/verification-email-template.ts(15,25): error TS2769: No overload matches this call.
  Overload 1 of 2, '(data: MailDataRequired, isMultiple?: boolean | undefined, cb?: ((err: Error | ResponseError, result: [ClientResponse, {}]) => void) | undefined): Promise<...>', gave the following error.`

**Comment by saveraa** [comment]
If this has changed in 7.0.0, where is latest documentation for the send api please? Can't find much here. @childish-sambino 

Thanks.

**Comment by childish-sambino** [comment]
Yes, this was released in [6.5.5](https://github.com/sendgrid/sendgrid-nodejs/releases/tag/6.5.5)

**Comment by isaachinman** [comment]
@childish-sambino What do you think about exporting `MailDataRequired` from the `@sendgrid/mail` package as well?

**Comment by childish-sambino** [comment]
@isaachinman I've no issue with that. If you want to submit a PR I can review it.

**Comment by isaachinman** [comment]
@childish-sambino I had a look around, and am unfamiliar with the module approach of `export =` being taken [here](https://github.com/sendgrid/sendgrid-nodejs/blob/master/packages/mail/index.d.ts#L3). I'm not clear on how this can be expanded without introducing breaking changes.

Happy to help if you can point me in the right direction.

**Comment by childish-sambino** [comment]
Think this line can just be updated: https://github.com/sendgrid/sendgrid-nodejs/blob/master/packages/mail/src/mail.d.ts#L37

Like this maybe:
```
declare const mail: MailService & { MailService: typeof MailService, MailDataRequired: typeof MailDataRequired };
```

**Comment by isaachinman** [comment]
No, that won't work as `MailDataRequired` is already a type. The use of `declare const` makes things slightly complicated.

**Comment by childish-sambino** [comment]
Then just `, MailDataRequired: MailDataRequired };`?

**Comment by isaachinman** [comment]
No, what I am saying is that literally adds a `MailDataRequired` attribute to the default export. If you tried out your latest suggestion, you'd get a `refers to a value, but is being used as a type` error.

I haven't seen any other packages managing types with this sort of module approach, so I can't really offer you any suggestions. You need a way of supporting named exports.

Normally I'd expect to see something like this:

```jsx
export default mail
export { MailDataRequired }
```

**Comment by childish-sambino** [comment]
@isaachinman Think this (hack) should fix it: https://github.com/sendgrid/sendgrid-nodejs/pull/1102

**Comment by yenicelik** [comment]
still having the same issue, and proper workaround?

**Comment by mkmrcodes** [comment]
Still same issue with version 7.6.0

**Comment by peterj** [comment]
Not perfect, but it allows you to go past the error:

```typescript
// @ts-expect-error: https://github.com/sendgrid/sendgrid-nodejs/issues/1057
await sgMail.sendMultiple(msg);
```

**Comment by HubbardJacob** [comment]
Was also having this issue with 7.6.0. Had to force compiler to ignore it.

**Comment by alzayatn** [comment]
Also having the same issue with version `7.6.1`

**Comment by childish-sambino** [comment]
This issue has been added to our internal backlog to be prioritized. Pull requests and +1s on the **issue summary** will help it move up the backlog.

**Comment by mgara** [comment]
Hello, any workaround for this ? thanks !
